### ISSUE TITLE
Make URLSlug on product optional

### ISSUE THREAD

**Issue by dkarlovi** [coreshop/CoreShop#2124]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no

Currently, the product slug is mandatory and created by Coreshop if not specified. This creates a lot of slugs, in turn creating problems, see #2025. Also, if you have a lot of products (we're doing a big demo site with 250k), the slugs table is *huge*.

All this for IMO very little benefit: the slug is generated by Coreshop and materialized many times. It could equally be matched by just having a single route like Coreshop 2 had, `coreshop_product` IIRC.

We could re-add the route and have it be used as a fallback. If the product specifies a slug, it's used, but if not, it's not used, the route URL is. This isn't just adding the route and call it a day because some parts of Coreshop now expect the slug to exist, if not they'll throw an exception.

**Comment by dpfaffenbauer** [comment]
@dkarlovi this is now possible

```
<?php

namespace App\CoreShop;

use CoreShop\Component\Pimcore\Slug\DataObjectSlugGeneratorInterface;
use CoreShop\Component\Pimcore\Slug\SluggableInterface;
use Pimcore\Model\DataObject\APimcoreDataObject;

class SlugGenerator implements DataObjectSlugGeneratorInterface
{
    public function __construct(private DataObjectSlugGeneratorInterface $generator)
    {
    }

    public function generateSlugs(SluggableInterface $sluggable): void
    {
        if ($sluggable instanceof APimcoreDataObject) {
            return;
        }

        $this->generator->generateSlugs($sluggable);
    }
}
``?

**Comment by dkarlovi** [comment]
What do you think about re-adding the product route as a fallback?

**Comment by dkarlovi** [comment]
What do you think about re-adding the product route as a fallback?

**Comment by dpfaffenbauer** [comment]
sure, fine

**Comment by dpfaffenbauer** [comment]
we could also make the slugs a configuration value to disable them even faster and easier

**Comment by dkarlovi** [comment]
Probably closed by #2143, could use a recipe how to prune the slugs table from the current (populated) state. Can it be safely truncated?

**Comment by dpfaffenbauer** [comment]
well, SEO wise no, CoreShop wise yes.
